PurchasesCoreSwift/SubscriberAttributes/AttributionDataMigrator.swift

```diff
@@ -0,0 +1,145 @@
+        }
+        let networkSpecificSubscriberAttributes: [String: Any]
+        switch network {
+        case .adjust:
+            networkSpecificSubscriberAttributes = convertAdjustAttribution(attributionData)
+        case .appsFlyer:
+            networkSpecificSubscriberAttributes = convertAppsFlyerAttribution(attributionData)
+        case .branch:
+            networkSpecificSubscriberAttributes = convertBranchAttribution(attributionData)
+        case .tenjin,
+             .facebook:
+            networkSpecificSubscriberAttributes = [:]
+        case .mParticle:
+            networkSpecificSubscriberAttributes = convertMParticleAttribution(attributionData)
+        case .none, .appleSearchAds:
+            // Apple Search Ads uses standard attribution system
+            networkSpecificSubscriberAttributes = [:]
+        }
+        return convertedAttribution.merging(networkSpecificSubscriberAttributes) { (_, new) -> Any? in
+            new
+        }
+    }
+
+}
+
+private extension AttributionDataMigrator {
+
+    func convertMParticleAttribution(_ data: [String: Any]) -> [String: Any] {
+        var convertedAttribution: [String: Any] = [:]
+        if let value = data[AttributionKey.MParticle.id.rawValue] {
+            convertedAttribution[SpecialSubscriberAttributes.mpParticleID] = value
+        }
+        if let value = data[AttributionKey.networkID.rawValue] {
+            convertedAttribution[SpecialSubscriberAttributes.mpParticleID] = value
+        }
+        return convertedAttribution
+    }
+
+    func convertBranchAttribution(_ data: [String: Any]) -> [String: Any] {
+        var convertedAttribution: [String: Any] = [:]
+        if let value = data[AttributionKey.Branch.channel.rawValue] {
+            convertedAttribution[SpecialSubscriberAttributes.mediaSource] = value
+        }
+        if let value = data[AttributionKey.Branch.campaign.rawValue] {
+            convertedAttribution[SpecialSubscriberAttributes.campaign] = value
+        }
+        return convertedAttribution
+    }
+
+    // swiftlint:disable cyclomatic_complexity
+    func convertAppsFlyerAttribution(_ data: [String: Any]) -> [String: Any] {
+        var fixedData = data
+        if let innerDataObject = fixedData[AttributionKey.AppsFlyer.dataKey.rawValue] as? [String: Any?] {
+            if fixedData[AttributionKey.AppsFlyer.statusKey.rawValue] != nil {
+                for (key, value) in innerDataObject {
+                    fixedData[key] = value
+                }
+            }
+        }
+
+        var convertedAttribution: [String: Any] = [:]
+
+        if let value = fixedData[AttributionKey.networkID.rawValue] {
+            convertedAttribution[SpecialSubscriberAttributes.appsFlyerID] = value
+        }
+        if let value = fixedData[AttributionKey.AppsFlyer.id.rawValue] {
+            convertedAttribution[SpecialSubscriberAttributes.appsFlyerID] = value
+        }
```

Comment:
I'm not familiar with how the mapping should go for these, so I'm assuming that they're correct, and that these are mostly coming from reverse-engineering the values, right? there's no documentation we can point to?

Reply:
I looked into khepri's code. I could point to that, but being this open source I am not sure it would make sense.

Reply:
I added a comment to the new extracted function